### PR TITLE
Resolves DENG-3134 to create a dataset for Braze ingestions

### PR DESCRIPTION
**Pull request by chelseybeck**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3197)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/braze_external/dataset_metadata.yaml
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/braze_external/dataset_metadata.yaml:9]: Maybe worth checking with Data SRE, but shouldn't this look more like this:
```
workgroup_access:
- role: roles/bigquery.dataViewer
   members:
   - workgroup:braze
- role: roles/bigquery.dataWriter
   members:
   - workgroup:braze/writers
```

?

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/braze_external/dataset_metadata.yaml:9]: I found an example:
https://github.com/mozilla/bigquery-etl/blob/8323733162134c27b6207080059a8f7ea49263e8/sql/moz-fx-data-shared-prod/search_terms_derived/sanitization_job_metadata_v2/metadata.yaml#L17-L23

Comment (whd) [reply sql/moz-fx-data-shared-prod/braze_external/dataset_metadata.yaml:9]: Kik's correct. We want read access for `workgroup:braze` and write access for `workgroup:braze/writers`.

Comment (chelseybeck) [reply sql/moz-fx-data-shared-prod/braze_external/dataset_metadata.yaml]: thanks @kik-kik & @whd! i've pushed those changes :)

**Review by whd — APPROVED** [review]
LGTM assuming the yaml lint passes

**Pull request merged**
